third-party/README.devel

```diff
@@ -0,0 +1,41 @@
+=========================================
+Chapel third-party developer instructions
+=========================================
+
+For certain third-party packages, currently gasnet, hwloc, and jemalloc,
```

Comment:
The list of current packages is very, very likely to become out of date, so I might just remove it, or say something more like "some examples include"